deploy/setup_server.yml

```diff
@@ -0,0 +1,95 @@
+- hosts: all
+  remote_user: vagrant
+  sudo: True
+
+  vars:
+    app_name: ifs
+    app_run: server
+    src_dir: /vagrant
+    srv_dir: /srv
+
+  tasks:
+    - name: install packages
+      apt: pkg={{item}} state=installed update-cache=yes
+      with_items:
+        - git-core
+        - npm
+        - nginx
+        - python3-pip
+        - python3-venv
+        - python3-dev
+        - supervisor
+        - postgresql
+
+    - name: ensure app directory exists
+      file: path={{srv_dir}}/{{app_name}} state=directory
+
+    - name: ensure log directory exists
+      file: path={{srv_dir}}/{{app_name}}/log state=directory
+
+    - name: link source directory
+      file: src={{src_dir}} dest={{srv_dir}}/{{app_name}}/src state=link
+
+    - name: install bower
+      npm: name=bower global=yes
+
+    - name: install grunt
+      npm: name=grunt global=yes
+
+    - name: install node
+      npm: name=node global=yes
+
+    - name: make node -> nodejs link
+      file: src=/usr/bin/nodejs dest=/usr/bin/node state=link
+
+    - name: install grunt-cli
+      npm: name=grunt-cli global=yes
+
+    - name: install clientside dependencies with npm
+      npm: path={{srv_dir}}/{{app_name}}/src
+
+    - name: install clientside dependencies with bower
+      bower: path={{srv_dir}}/{{app_name}}/src
```

Comment:
I'm getting an error here when running ``vagrant provision`` with the latest code. I'm having some difficulties debugging it. Seems like it's having some trouble decoding a JSON file from ``bower.list()``

```
TASK: [install clientside dependencies with bower] **************************** 
failed: [localhost] => {"failed": true, "parsed": false}
BECOME-SUCCESS-ucubgguacvmjsqycvosqnoqzplysownz
Traceback (most recent call last):
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1443723918.53-268417569323185/bower", line 1800, in <module>
    main()
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1443723918.53-268417569323185/bower", line 168, in main
    installed, missing, outdated = bower.list()
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1443723918.53-268417569323185/bower", line 116, in list
    data = json.loads(self._exec(cmd, True, False))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
OpenSSH_7.1p1, OpenSSL 1.0.2d 9 Jul 2015
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: auto-mux: Trying existing master
debug1: mux_client_request_session: master session id: 2
Shared connection to 127.0.0.1 closed.


FATAL: all hosts have already failed -- aborting
```